### PR TITLE
refactor: remove unnecessary object._rawDup() template

### PR DESCRIPTION
**Pull request by WalterBright**
Reduce the thicket of templates in object.d

### CODE REVIEW

File: src/object.d [hunk 1]
```diff
@@ -3355,28 +3355,30 @@ private U[] _dup(T, U)(scope T[] a) // pure nothrow depends on postblit
...
     {
-        U[] res;
-        foreach (ref e; a)
-            res ~= e;
-        return res;
+        static if (is(T : void))
+            assert(0, "Cannot dup a void[] array at compile time.");
+        else
+        {
+            U[] res;
```
Comment (andralex) [inline src/object.d]: This code is uncovered.

File: src/object.d [hunk 2]
```diff
@@ -3355,28 +3355,30 @@ private U[] _dup(T, U)(scope T[] a) // pure nothrow depends on postblit
...
+            foreach (ref e; a)
+                res ~= e;
```
Comment (andralex) [inline src/object.d]: aw, this is quite inefficient, we should look at this soonish

**Comment by andralex** [comment]
OK but you should know that the net number of line actually increases...

File: src/object.d (hunk 1 above)
Comment (WalterBright) [inline src/object.d]: Yes, but it was before, as this is a refactoring.

File: src/object.d (hunk 2 above)
Comment (WalterBright) [inline src/object.d]: That branch is only executed when doing ctfe.

**Pull request merged**